pkg/api/norman/customization/cluster/validator.go

```diff
@@ -248,6 +248,10 @@ func (v *Validator) validateK3sBasedVersionUpgrade(request *types.APIContext, sp
 		return err
 	}
 
+	if (isK3s && cluster.Spec.K3sConfig == nil) || (isrke2 && cluster.Spec.Rke2Config == nil) {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, "k3sConfig cannot be changed from nil")
```

Comment:
Should we separate these checks so the error message corresponds to the field that was changed?

Reply:
Actually, embedded cluster is k3s only so I will remove rke check.